clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesCredentials.java

```diff
@@ -28,21 +28,27 @@
 import java.util.*;
 
 public class KubernetesCredentials {
+  private final String name;
   private final KubernetesApiAdaptor apiAdaptor;
   private final List<String> namespaces;
   private final List<LinkedDockerRegistryConfiguration> dockerRegistries;
   private final HashMap<String, List<String>> imagePullSecrets;
   private final Logger LOG;
+  private final List<String> requiredGroupMembership;
 
-  public KubernetesCredentials(KubernetesApiAdaptor apiAdaptor,
+  public KubernetesCredentials(String name,
+                               KubernetesApiAdaptor apiAdaptor,
                                List<String> namespaces,
                                List<LinkedDockerRegistryConfiguration> dockerRegistries,
-                               AccountCredentialsRepository accountCredentialsRepository) {
+                               AccountCredentialsRepository accountCredentialsRepository,
+                               List<String> requiredGroupMembership) {
```

Comment:
I went through the whole refactoring of the `GoogleCredentials` object(s) because of this line - I don't think `requiredGroupMembership` belongs in <provider>Credentials.

The way the code is structured, the <provider>NamedAccountCredentials is supposed to hold the surrounding metadata + accessor objects (`Compute` in the GCE case, `KubernetestApiAdaptor` in K8s), while also allowing direct access to the _real_ credentials (For AWS its the access key + secret key, for GCE it's the contents of the JSON credentials file). What's the equivalent for K8s?